[SamsonChew] iP

SamsonChew opened the pull request:
## SamsonXiaoyi
`Your best AI assistant`

SamsonXiaoyi frees your mind of having to remember things you need to do. It's,

- text-based
* easy to learn
+ ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here]()
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 💯 

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

SamsonChew committed:
Rename,Greet,Exit

SamsonChew committed:
Echo

SamsonChew committed:
Add, List

SamsonChew committed:
Mark as Done

SamsonChew committed:
ToDos,Events,Deadlines

SamsonChew committed:
Automated Text UI Testing

SamsonChew committed:
Handle Errors

SamsonChew committed:
Delete

SamsonChew committed:
Use Enums

SamsonChew committed:
Samson Level-7: Save

SamsonChew committed:
Merge branch 'branch-Level-7'

Adds Level-7: Save

Save the tasks in the hard disk automatically
whenever the task list changes

SamsonChew committed:
Adds Level-8: Dates and Times

SamsonChew committed:
Merge branch 'branch-Level-8'
Teach the chatbox how to understand dates and times

SamsonChew committed:
Samson A-MoreOOP: Use More OOP

SamsonChew committed:
Merge branch 'branch-A-MoreOOP'

A-MoreOOP: Use More OOP

Refactor the code to extract out closely related code as classes

SamsonChew committed:
A-Packages:Organize into Packages

SamsonChew committed:
Merge branch 'branch-A-Packages'

A-Packages: Organize into Packages

Divide classes into packages

SamsonChew committed:
Merge branch 'add-gradle-support'

Merge add-gradle-support to master

SamsonChew committed:
A-Gradle: Use Gradle

SamsonChew committed:
Add A-JUnit: Add JUnit Tests

SamsonChew committed:
Merge branch 'branch-A-JUnit'
Add JUnit Tests

Add JUnit tests to test the behavior of the code

SamsonChew committed:
Create a JAR File

SamsonChew committed:
Add JavaDoc comments A-JavaDoc

SamsonChew committed:
Merge branch 'branch-A-JavaDoc'

Add JavaDoc comments

Add JavaDoc comments to the code

SamsonChew committed:
Follow the Coding Standard

SamsonChew committed:
Merge branch 'branch-A-CodingStandard'
Follow the Coding Standard

Tweark the code to comply with a coding standard

SamsonChew committed:
Add Find feature

SamsonChew committed:
Merge branch 'branch-Level-9'

Add Find Feature

Give users a way to find a task by searching
for a keyword in the task description

EthanLoww commented on src/main/java/samson/Parser.java:
```diff
@@ -0,0 +1,103 @@
+package samson;
+
+import samson.command.*;
```
Imported classes should always be listed explicitly

SamsonChew replied:
Thanks for pointing this down will ammend on it

EthanLoww commented on src/main/java/samson/Parser.java:
```diff
@@ -0,0 +1,103 @@
+                            "Example call: todo your_task");
+                }
+                return new AddCommand(new ToDo(words[1].trim()));
+            case "deadline":
+                if (words.length < 2 || words[1].trim().isEmpty()) {
+                    throw new SamException("Please provide the description and deadline of your task \n " +
+                            "Example call: deadline your_task /by yyyy-MM-dd HHmm");
+                }
+                String[] deadlineParts = words[1].split(" /by ");
+                if (deadlineParts.length < 2) {
+                    throw new SamException("Please provide the deadline using /by.");
+                }
+                String deadlineDescription = deadlineParts[0].trim();
+                String by = deadlineParts[1].trim();
+                return new AddCommand(new Deadline(deadlineDescription, by));
+            case "event":
+                if (words.length < 2 || words[1].trim().isEmpty()) {
+                    throw new SamException("The description, start, and end time of an event cannot be empty. \n " +
+                            "Example call: event your_event /from yyyy-MM-dd HHmm /to yyyy-MM-dd HHmm");
+                }
+                String[] eventParts = words[1].split(" /from | /to ");
+                if (eventParts.length < 3) {
+                    throw new SamException("Please provide the event timings using /from and /to.");
+                }
+                String eventDescription = eventParts[0].trim();
+                String from = eventParts[1].trim();
+                String to = eventParts[2].trim();
+                return new AddCommand(new Event(eventDescription, from, to));
+            case "find":
+                if (words.length < 2 || words[1].trim().isEmpty()) {
+                    throw new SamException("Please provide a keyword to search for. \n" +
+                            "Example call: find your_task ");
+                }
+                return new FindCommand(words[1].trim());
+            default:
+                throw new SamException("I'm sorry, but I don't know what that means :-( \n " +
+                        "Kindly provide the tasks starting with 'todo', 'event', 'deadline'!!");
+        }
+    }
+}
```
For switch statements, there should not be any indentation for case clauses.
Method is too long, consider using helper methods to handle command creation

SamsonChew replied:
Thanks for pointing it out, have amended it accordingly for switch statement indentation error

EthanLoww commented on src/main/java/samson/Storage.java:
```diff
@@ -0,0 +1,119 @@
+                default:
+                    System.out.println("Invalid data detected: " + line);
+                    break;
+            }
```
Again, case clauses should not be indented

SamsonChew replied:
Thank you for pointing it out, have amended accordingly.

EthanLoww commented on src/main/java/samson/command/AddCommand.java:
```diff
@@ -0,0 +1,50 @@
+
+import samson.Storage;
+import samson.Ui;
+import samson.task.*;
```
Imported classes should be listed out explicitly

SamsonChew replied:
Amended. Thank you.

EthanLoww commented on src/test/java/samson/ParserTest.java:
```diff
@@ -0,0 +1,89 @@
+package samson;
+
+import org.junit.jupiter.api.Test;
+import samson.command.*;
```
Imported classes should be listed explicitly

SamsonChew replied:
Thank you. Has solved.

EthanLoww reviewed (commented):
Looks good overall, just a few coding standard violations👍.

ttzuche commented on src/main/java/samson/Parser.java:
```diff
@@ -5,9 +5,20 @@
+ * The  class is responsible for interpreting user input and converting
+ * it into executable commands. It parses the input string and returns the corresponding
+ * <code> Command </code> object, which can be executed to perform the desired action.
+ */
```
The Javadocs were well-written and clear for all classes and methods, which makes the understanding of code better.

ttzuche commented on src/main/java/samson/SamException.java:
```diff
@@ -1,7 +1,17 @@
 package samson;
```
Since `SamException` is an exception class, I think it would be better to create another package called exception inside of the samson package and put this class inside that package.

ttzuche commented on src/main/java/samson/Ui.java:
```diff
@@ -5,44 +5,78 @@
         System.out.println("____________________________________________________________");
         System.out.println(" Bye. Hope to see you again soon!");
         // System.out.println("____________________________________________________________");
     }
```
Maybe you can close your scanner in this function?

ttzuche commented on src/main/java/samson/Storage.java:
```diff
@@ -40,15 +61,15 @@ public List<Task> loadTaskFromFile() throws IOException, SamException {
-                    if (deadline.isDone()) {
+                    if (isDone) {
                         deadline.complete();
                     }
```
The indentation for switch case is incorrect, there is no need to indent for 'case' branches.

ttzuche reviewed (commented):
Overall the code looks great and easy to understand! 👍
Javadocs are clear. Just need to fix some indentation.

SamsonChew committed:
Fix minor bug and coding standard

SamsonChew committed:
Add CheckStyle feature

SamsonChew committed:
Merge branch 'A-CheckStyle'

Add CheckStyle feature

Use checkStyle to detect coding style violations

SamsonChew committed:
Add GUI to chatbot

SamsonChew committed:
Merge branch 'branch-Level-10'
Add GUI to chatbot

Use JavaFX technology to implement the GUI

SamsonChew committed:
Add assertions to TaskList operations

Use assert feature (not JUnit assertions) to document important assumptions that should hold at various points in the code.

To validate the task index to prevent over-indexing.

SamsonChew committed:
Adjust Coding Standard  Remove too-long method to increase the readability of the code

SamsonChew committed:
Merge pull request #1 from SamsonChew/branch-A-Assertions

Add assertions to TaskList operations

SamsonChew committed:
Merge branch 'master' into branch-A-CodeQuality
Adjust Coding Standard

Refactor too-long methods into different methods
to improve the readability of code

SamsonChew committed:
Merge pull request #2 from SamsonChew/branch-A-CodeQuality

Adjust Coding Standard

SamsonChew committed:
Add flexibility in searching for items

Reimplement searching algorithm to find items
even if the keyword matches the item only partially

Create more effective searching even without
complete inputs

SamsonChew committed:
Merge branch 'branch-C-BetterSearch'
Add flexibility in searching for items

Reimplement searching algorithm to find items
even if the keyword matches the item only partially

Create more effective searching even without
complete inputs

SamsonChew committed:
Merge branch 'master' of https://github.com/SamsonChew/ip
Resolve merge conflict

SamsonChew committed:
Change boolean variable naming

SamsonChew committed:
Remove too-long method

SamsonChew committed:
Add Ui page screenshot

SamsonChew committed:
Update user guide for version 2.0

SamsonChew committed:
Update UiTest file for extra test
